packages/commonwealth/server/api/external-router-middleware.ts

```diff
@@ -32,48 +38,73 @@ export async function apiKeyAuthMiddleware(
   if (typeof addressHeader !== 'string')
     throw new AppError('Unauthorized', 401);
 
-  const addressInstance = await models.Address.findOne({
-    attributes: ['user_id'],
-    where: {
-      address: addressHeader,
-      verified: { [Op.ne]: null },
-    },
-    include: [
+  let user: z.infer<typeof User> | undefined;
+  const salt = await cache().getKey(
+    CacheNamespaces.Api_key_auth,
+    buildApiKeySaltCacheKey(addressHeader),
```

Comment:
looks like two calls to redis, one for salt and one for user info... maybe one kvp by address->user + salt is better? ... I've seen hash+salt stored and cached together